examples/with-react-query-typescript/hooks/usePosts.ts

```diff
@@ -0,0 +1,26 @@
+import { TPost } from '../types';
+import {
+  useQuery,
+  UseQueryResult,
+} from "react-query";
+
+export const api = async<T>(resource: string): Promise<T>=> {
```

Comment:
I would move the TPost here.

```typescript
export const api = async (resource: string): Promise<TPost>=> {
``` 

In `usePosts` I would do the same (when using explicit return we don't rely on useQuery inference).

```typescript
export const usePosts = (key: any, url: string, isRouterReady: boolean): UseQueryResult<TPost, unknown> => {
```

With this change, consumers does not have to decide which type to set

```typescript
const { isLoading, isError, data } = usePosts(
```

Reply:
`api` is a generic function. 

We are using it here for posts only, but it can be used for any type. 

In real world,`api` should have been in `lib` directory or something. But for simplicity, I kept it inside `usePosts`.

Reply:
So generic api make sense yes.

For the usePost, I would create two versions with proper parameters and return types (remove key:any at the same time).

I would find it more explicit when using them.

Reply:
As suggested, removed generic types from `usePosts`.